File: chapter-7/python/ex_7_1.py
# Programming Exercise 7-1
#
# Program to Total a week's sales (seven inputs).
# This program prompts a user for a sales amount for each day of the week,
# totals them up, then displays the total formatted as currency.
from validations import prompt_for_float


# Define the main function
def main():
    # Define local float variable for rtotal sales


    # Initialize lists for daily sales and days of the week (7 elements each)
    daily_sales = [0.0 for i in range(7)]
    days_of_week = ["Sunday",
                    "Monday",
                    "Tuesday",
                    "Wednesday",
                    "Thursday",
                    "Friday",
                    "Saturday"]

    # loop 7 times
    for i in range(7):
        prompt_message = "Enter sales for {}:".format(days_of_week[i])
        daily_sales[i] = prompt_for_float(prompt_message, "That's not a number")

    # Display total sales; use a $ and format the value to two decimal places
    print("Total Sales: ${:.2f}".format(sum(daily_sales)))





# Call the main function to start the program
main()

File: chapter-7/elixir/ex_7_1.exs
defmodule Exercise1 do

  def prompt_for_float(prompt_text, invalid_message) do
    potential_value = IO.gets(prompt_text) |> Float.parse
    case potential_value do
      {value,_} -> value
      :error -> 
        IO.puts invalid_message
        prompt_for_float(prompt_text, invalid_message)
    end
  end

  def format_float(x) do
    [formatted|rest] = :io_lib.format("~.2f~n",[x])
    formatted
  end

  def main() do
    days_of_week = ~w(Sunday Monday Tuesday Wednesday Thursday Friday Saturday)
    total_sales =
      days_of_week
      |> Enum.map(fn(day) -> "Enter sales for #{day}: " end)
      |> Stream.map(&(prompt_for_float(&1, "Not a number")))
      |> Enum.sum
      |> format_float
    IO.puts "Total Sales: #{total_sales}"
  end

end



Exercise1.main()
